controller.go

```diff
@@ -270,6 +284,22 @@ func (c *controller) NewNetwork(networkType, name string, options ...NetworkOpti
 
 	network.processOptions(options...)
 
+	if !c.started {
+		// Global stores are watched only when the controller enters the 'started' state.
+		// Hence, its expected that we could possibly get a newNetwork create call only
+		// for locally scoped networks
+		n := c.getNetworkByNameFromLocalStore(name)
+		if n != nil {
+			n.generic = network.generic
+			n.endpoints = network.endpoints
```

Comment:
Pls remove updating `n.endpoints` here in your latest commit. This is not being done in `setupNetwork` which is common for multiple codepath.